dxr/static/templates/path_line.html

```diff
@@ -1,10 +1,13 @@
 {%- if not is_first_or_only -%}
-<span class="path-separator">/</span>
+    <span class="path-separator">/</span>
 {%- endif -%}
 {%- if not is_binary or is_dir -%}
-<a href="{{ url }}" {%- if is_dir %} data-path="{{ data_path }}" {% endif %}>
+    <a href="{{ url }}" {%- if is_dir %} data-path="{{ data_path }}" {% endif %}>
 {%- endif -%}
-    {{ display_path }}
+{# Remark: it is possible for <b> tags to be unbalanced within fragments if it opens in one
+   fragment and closes in another, but we avoid any issue by showing the entire path as a
+   single element. #}
+{{ display_path|safe }}
```

Comment:
Since we need to be sure we trust the input when we use `|safe`, I have traced execution path from request to template here: https://gist.github.com/pelmers/7d53a0fa02364f84542c.
In particular I think the portion to focus on is lines [15-34](https://gist.github.com/pelmers/7d53a0fa02364f84542c#file-xsstrace-py-L15-L34), where at the bottom I explain my reasoning for why I believe that the approach is safe.